### DISCUSSION TITLE
barButtonItemの画像を変更したい

### DISCUSSION THREAD

**Discussion by oyui7272** [lit-iphone-course/lit-iphone-course#53] [Question-original] [answered]
### (1) 実現したいこと
SearchViewControllerの105行目あたりで設定している変数transportationの値によってnavigationBarButtonItemの画像を変更したいです

### (2) 実現するために自分が試した内容とその結果
こちらの記事を参考に書いたのですが、変更されませんでした
https://qiita.com/iduchikun/items/7eafb7413fb85648fb2f

`let newLeftBarButtonItem = UIBarButtonItem(image: UIImage(named: img_url), style: .plain, target: self, action: .none)`
`self.navigationItem.leftBarButtonItem = newLeftBarButtonItem`
※ img_urlにはtransportationによって変わる画像のurlが入ってます

### (3) 調べた内容や資料
上の記事と同じようなサイトをいくつか調べたのですが、いずれも同じような書き方をしていました！
https://mgre.co.jp/blog/9645

### (4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
navigationControllerを使ってないことも原因の一つなのかもしれない(?)と考えています

### (5) GitHubのリンク
https://github.com/yuipi7272/NanoTour.git

**Comment by txxxxc** [comment, marked as answer]
色々調べてみた感じ、NavigationControllerを繋げずにNavigationItemを使っている関係でうまくコードから変更できてないっぽいので、以下の画像のようにTabBarControllerにNavigationControllerをつなげてあげてその上にViewControllerを乗せる感じでやるとうまくいくと思いますー！
![image](https://user-images.githubusercontent.com/22112440/192158693-9340027a-7cbd-40d3-99bd-9a6574e0b138.png)

**Reply by oyui7272** [reply]
ありがとうございます！！できました😭🙏
